akka-persistence-tck/src/main/scala/akka/persistence/PluginSpec.scala

```diff
@@ -35,10 +46,87 @@ abstract class PluginSpec(val config: Config) extends TestKitBase with WordSpecL
 
   def extension: Persistence = _extension
 
+  /**
+   * Returns a new unique persistence id
+   */
+  def nextPid: String =
+    s"p-${counter.incrementAndGet()}"
+
+  /**
+   * Returns the current persistence id
+   */
   def pid: String = _pid
 
+  /**
+   * Returns the current writer uuid
+   */
   def writerUuid: String = _writerUuid
 
+  /**
+   * Returns the actorRef of the default configured journal plugin
+   */
+  def journal: ActorRef =
+    extension.journalFor(null)
+
   def subscribe[T: ClassTag](subscriber: ActorRef) =
     system.eventStream.subscribe(subscriber, implicitly[ClassTag[T]].runtimeClass)
+
+  /**
+   * Overridable hook that is called before populating the journal for the next
+   * test case. `pid` is the `persistenceId` that will be used in the test.
+   * This method may be needed to clean pre-existing events from the log.
+   */
+  def preparePersistenceId(pid: String): Unit = ()
+
+  /**
+   * Implementation may override and return false if it does not
+   * support atomic writes of several events, as emitted by `persistAll`.
+   */
+  def supportsAtomicPersistAllOfSeveralEvents: Boolean = true
+
+  def replayedMessage(snr: Long, deleted: Boolean = false, confirms: Seq[String] = Nil): ReplayedMessage =
```

Comment:
Would 'protected' be better than private as they should be overridden by implementations?